src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -522,6 +594,97 @@ long TriangleCountExecutor::getTriangleCount(int graphId, std::string host, int
     }
 }
 
+bool TriangleCountExecutor::proceedOrNot(std::set<string> partitionSet,int partitionId) {
+
+    const std::lock_guard<std::mutex> lock(aggregateWeightMutex);
+
+    //triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Aquired ", "info");
+    std::set<string>::iterator partitionSetIterator;
+    std::map<int,int> tempWeightMap;
+    for (partitionSetIterator = partitionSet.begin(); partitionSetIterator != partitionSet.end(); ++partitionSetIterator) {
+        std::string partitionIdString = *partitionSetIterator;
+        int currentPartitionId = atoi(partitionIdString.c_str());
+        tempWeightMap[currentPartitionId] = aggregateWeightMap[currentPartitionId];
+    }
+
+    int currentWorkerWeight = tempWeightMap[partitionId];
+    pair<int, int> entryWithMinValue = make_pair(partitionId,currentWorkerWeight);
+
+    map<int, int>::iterator currentEntry;
+
+    for (currentEntry = aggregateWeightMap.begin(); currentEntry != aggregateWeightMap.end(); ++currentEntry) {
+        if (entryWithMinValue.second > currentEntry->second) {
+            entryWithMinValue = make_pair(currentEntry->first,currentEntry->second);
+        }
+    }
+
+    if (entryWithMinValue.first == partitionId) {
+        int currentWeight = aggregateWeightMap[entryWithMinValue.first];
+        currentWeight++;
+        aggregateWeightMap[entryWithMinValue.first] = currentWeight;
+        triangleCount_logger.log("###COMPOSITE### Aggregator Initiated : Partition ID: " + std::to_string(partitionId) + " Weight : " + std::to_string(currentWeight), "info");
+        return true;
+    } else {
+        return false;
+    }
+
+
+
+    aggregateWeightMutex.unlock();
+    //triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Released ", "info");
+}
+
+bool TriangleCountExecutor::updateMap(int partitionId) {
+    //aggregateWeightMutex.lock();
+    const std::lock_guard<std::mutex> lock(aggregateWeightMutex);
+
+    //triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Aquired on completion", "info");
+    int currentWeight = aggregateWeightMap[partitionId];
+    currentWeight--;
+    aggregateWeightMap[partitionId] = currentWeight;
+    triangleCount_logger.log("###COMPOSITE### Aggregator Completed : Partition ID: " + std::to_string(partitionId) + " Weight : " + std::to_string(currentWeight), "info");
+    //triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Released on completion", "info");
+}
+
+int TriangleCountExecutor::updateTriangleTreeAndGetTriangleCount(std::vector<std::string> triangles) {
+
+    const std::lock_guard<std::mutex> lock1(triangleTreeMutex);
+    std::vector<std::string>::iterator triangleIterator;
+    int aggregateCount = 0;
+
+    triangleCount_logger.log("###COMPOSITE### Triangle Tree locked ", "info");
```

Comment:
This kind of logs should be in debug level

Reply:
Changed the log level

Reply:
Ok